programmers/Recommend-a-new-ID.js

```diff
@@ -0,0 +1,30 @@
+function solution(new_id) {
+    new_id = new_id.toLowerCase(); // 1단계
+
+    new_id = new_id.replace(/[^0-9a-z-_.]/g, ""); // 2단계
+
+    new_id = new_id.replace(/\.{2,}/g, "."); // 3단계
+
+    new_id = new_id.replace(/^\./, "");
+    new_id = new_id.replace(/\.$/, "");      // 4단계
```

Comment:
같은 변수는 붙여주는게 보기 더 좋을꺼 같아요
```suggestion
    new_id = new_id.toLowerCase(); // 1단계
    new_id = new_id.replace(/[^0-9a-z-_.]/g, ""); // 2단계
    new_id = new_id.replace(/\.{2,}/g, "."); // 3단계
    new_id = new_id.replace(/^\./, "");
    new_id = new_id.replace(/\.$/, "");      // 4단계
```